fastavro/json_writer.py

```diff
@@ -0,0 +1,130 @@
+import json
+
+from .schema import extract_record_type, parse_schema
+from .six import iteritems, btou
+from .validation import validate
+
+
+def _write_json(datum, schema):
+    record_type = extract_record_type(schema)
+
+    passthrough_schemas = (
+        'null',
+        'boolean',
+        'string',
+        'int',
+        'long',
+        'float',
+        'double',
+        'enum',
+    )
+
+    if record_type in passthrough_schemas:
+        return datum
+
+    elif record_type == 'bytes':
+        return btou(datum, encoding='iso-8859-1')
+
+    elif record_type == 'fixed':
+        return btou(datum, encoding='iso-8859-1')
+
+    elif record_type == 'union':
+        best_match_index = -1
+        most_fields = -1
+        for index, candidate in enumerate(schema):
+            if validate(datum, candidate, raise_errors=False):
+                if extract_record_type(candidate) == 'record':
+                    fields = len(candidate['fields'])
+                    if fields > most_fields:
+                        best_match_index = index
+                        most_fields = fields
+                else:
+                    best_match_index = index
+                    break
+        if best_match_index < 0:
+            pytype = type(datum)
+            msg = '%r (type %s) do not match %s' % (datum, pytype, schema)
+            raise ValueError(msg)
+
+        best_match_schema = schema[best_match_index]
+
+        if best_match_schema == 'null':
+            return None
+        else:
+            best_records_type = extract_record_type(best_match_schema)
+            if best_records_type in ('record, enum, fixed'):
```

Comment:
```suggestion
            if best_records_type in ('record', 'enum', 'fixed'):
```
I believe this was the intent.

Reply:
Yes, definitely. I should put in a test that would have caught this.